packit/cli/prepare_sources.py

```diff
@@ -0,0 +1,125 @@
+from packit.config import pass_config, get_context_settings
+from packit.schema import JobConfigSchema
+
+logger = logging.getLogger("packit")
+
+
+def load_job_config(job_config):
+    if job_config:
+        try:
+            return JobConfigSchema().loads(job_config)
+        except Exception as ex:
+            logger.error(f"Loading of JobConfig wasn't successful: {ex}")
+    return None
+
+
+@click.command("prepare-sources", context_settings=get_context_settings())
+@click.option(
+    "--result-dir",
+    metavar="DIR",
+    help="Copy the sources into DIR. By default, `prepare_sources_result` directory "
+    "in the current working directory is created.",
+)
+@click.option(
+    "--upstream-ref",
+    default=None,
+    help="Git ref of the last upstream commit in the current branch "
+    "from which packit should generate patches "
+    "(this option implies the repository is source-git).",
+)
+@click.option(
+    "--bump/--no-bump",
+    default=True,
+    help="Specifies whether to bump version or not.",
+)
+@click.option(
+    "--release-suffix",
+    default=None,
+    type=click.STRING,
+    help="Specifies release suffix. Allows to override default generated:"
+    "{current_time}.{sanitized_current_branch}{git_desc_suffix}",
+)
+@click.option(
+    "--job-config",
+    default=None,
+    type=click.STRING,
+    hidden=True,
+    help="Overrides package config found in the repository (needed for packit service).",
+)
+@click.option(
+    "--ref",
+    default="",
+    type=click.STRING,
+    help="Git reference to checkout.",
+)
+@click.option(
+    "--pr-id",
+    default=None,
+    type=click.STRING,
+    help="Specifies PR to checkout.",
+)
+@click.option(
+    "--merge-pr/--no-merge-pr",
+    is_flag=True,
+    default=True,
+    help="Specifies whether to merge PR into the base branch in case pr-id is specified.",
+)
+@click.argument(
+    "path_or_url",
+    type=LocalProjectParameter(
+        ref_param_name="ref",
+        pr_id_param_name="pr_id",
+        merge_pr_param_name="merge_pr",
+    ),
+    default=os.path.curdir,
+)
+@pass_config
+@cover_packit_exception
+def prepare_sources(
+    config,
+    path_or_url,
+    job_config,
+    upstream_ref,
+    bump,
+    release_suffix,
+    result_dir,
+    ref,
+    pr_id,
+    merge_pr,
+):
+    """
+    Prepare sources for a new SRPM build using content of the upstream repository.
+    Determine version, create an archive or download upstream and create patches for sourcegit,
+    fix/update the specfile to use the right archive, download the remote sources.
+    Behaviour can be customized by specifying actions (post-upstream-clone, get-current-version,
+    create-archive, create-patches, fix-spec-file) in the configuration.
+
+    PATH_OR_URL argument is a local path or a URL to the upstream git repository,
+    it defaults to the current working directory
+    """
+    job_config = load_job_config(job_config)
```

Comment:
I can see now this is indeed fairly tricky - I understand the use case is that people have multiple build options where they override different global variables to achieve multiple SRPM builds, such as leapp-repository:

https://github.com/oamg/leapp-repository/blob/master/.packit.yaml#L34

I guess the question is if we want to pass the actual, encoded job config to this command or just the index of the job within the config (or something else).

Personally, I'd pick the index route because shell-encoding json won't be pretty.

What are your thoughts, Laura?

Reply:
I have been thinking about using only an index in the beginning, but I came to the conclusion it wouldn't make things prettier (even though I agree that dumping and loading aren't nice as well). This way it is explicit, what we want to use instead of package config,  also passing the index may look really weird. But I can look into whether after manipulating with the JobConfig in packit-service it would be even possible to get the index of the job in the package config easily. Currently, I am more for using directly job config.

Reply:
okay, I respect your decision :) we can always implement the other one as well; looking forward to trying this on stage!

Reply:
> we can always implement the other one as well; looking forward to trying this on stage!

sure, since this decision doesn't affect anything at the moment (it isn't used yet), it can be adjusted when I found out it doesn't work well with the script implementation